core/src/main/scala/cats/data/NonEmptyList.scala

```diff
@@ -0,0 +1,193 @@
+
+  /**
+   * Left-associative fold on the structure using f.
+   */
+  def foldLeft[B](b: B)(f: (B, A) => B): B =
+    tail.foldLeft(f(b, head))(f)
+
+  /**
+   * Right-associative fold on the structure using f.
+   */
+  def foldRight[B](lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    Foldable[List].foldRight(toList, lb)(f)
+
+  /**
+   * Left-associative reduce using f.
+   */
+  def reduceLeft(f: (A, A) => A): A =
+    tail.foldLeft(head)(f)
+
+  def coflatMap[B](f: NonEmptyList[A] => B): NonEmptyList[B] = {
+    @tailrec def consume(as: List[A], buf: ListBuffer[B]): List[B] =
+      as match {
+        case Nil => buf.toList
+        case a :: as => consume(as, buf += f(NonEmptyList(a, as)))
+      }
+    NonEmptyList(f(this), consume(this.tail, ListBuffer.empty))
+  }
+
+  def ===(o: NonEmptyList[A])(implicit A: Eq[A]): Boolean =
+    (this.head === o.head) && this.tail === o.tail
+
+  def show(implicit A: Show[A]): String =
+    toList.iterator.map(A.show).mkString("NonEmptyList(", ", ", ")")
+
+  override def toString: String = s"NonEmpty$toList"
+}
+
+object NonEmptyList extends NonEmptyListInstances {
+  def apply[A](head: A, tail: A*): NonEmptyList[A] = NonEmptyList(head, tail.toList)
+
+  def fromList[A](l: List[A]): Option[NonEmptyList[A]] =
+    if (l.isEmpty) None else Some(NonEmptyList(l.head, l.tail))
+
+  def fromListUnsafe[A](l: List[A]): NonEmptyList[A] =
+    if (l.nonEmpty) NonEmptyList(l.head, l.tail)
+    else throw new IllegalArgumentException("Cannot create NonEmptyList from empty list")
+}
+
+private[data] sealed trait NonEmptyListInstances extends NonEmptyListInstances0 {
+
+  implicit val catsDataInstancesForNonEmptyList: SemigroupK[NonEmptyList] with Reducible[NonEmptyList]
+      with Comonad[NonEmptyList] with Traverse[NonEmptyList] with MonadRec[NonEmptyList] =
+    new NonEmptyReducible[NonEmptyList, List] with SemigroupK[NonEmptyList]
+        with Comonad[NonEmptyList] with Traverse[NonEmptyList] with MonadRec[NonEmptyList] {
+
+      def combineK[A](a: NonEmptyList[A], b: NonEmptyList[A]): NonEmptyList[A] =
+        a concat b
+
+      override def split[A](fa: NonEmptyList[A]): (A, List[A]) = (fa.head, fa.tail)
+
+      override def reduceLeft[A](fa: NonEmptyList[A])(f: (A, A) => A): A =
+        fa.reduceLeft(f)
+
+      override def map[A, B](fa: NonEmptyList[A])(f: A => B): NonEmptyList[B] =
+        fa map f
+
+      def pure[A](x: A): NonEmptyList[A] =
+        NonEmptyList(x, List.empty)
+
+      def flatMap[A, B](fa: NonEmptyList[A])(f: A => NonEmptyList[B]): NonEmptyList[B] =
+        fa flatMap f
+
+      def coflatMap[A, B](fa: NonEmptyList[A])(f: NonEmptyList[A] => B): NonEmptyList[B] =
+        fa coflatMap f
+
+      def extract[A](fa: NonEmptyList[A]): A = fa.head
+
+      def traverse[G[_], A, B](fa: NonEmptyList[A])(f: (A) => G[B])(implicit G: Applicative[G]): G[NonEmptyList[B]] =
+        G.map2Eval(f(fa.head), Always(Traverse[List].traverse(fa.tail)(f)))(NonEmptyList(_, _)).value
```

Comment:
why not have a `.traverse` method on `NonEmptyList`? Why is this different from the others?